lib/src/simulator.dart

```diff
@@ -286,15 +291,17 @@ class Simulator {
     }
 
     while (hasStepsRemaining() &&
-        _exception == null &&
+        _simExceptions.isEmpty &&
         !_simulationEndRequested &&
         (_maxSimTime < 0 || _currentTimestamp < _maxSimTime)) {
       await tick();
     }
 
-    if (_exception != null) {
-      logger.severe(_exception.toString(), _exception, _stacktrace);
-      throw _exception!;
+    if (_simExceptions.isNotEmpty) {
```

Comment:
No need to check if it's not empty, since you're already looping through a list which may be empty